File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/infrastructure/CreatorStatisticsRepository.java
package com.eager.questioncloud.core.domain.creator.infrastructure;

import com.eager.questioncloud.core.domain.creator.model.CreatorStatistics;
import java.util.List;

public interface CreatorStatisticsRepository {
    void save(CreatorStatistics creatorStatistics);

    void saveAll(List<CreatorStatistics> creatorStatistics);

    CreatorStatistics findByCreatorId(Long creatorId);

    void deleteAllInBatch();
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/infrastructure/repository/CreatorStatisticsRepository.kt
package com.eager.questioncloud.core.domain.creator.infrastructure.repository

import com.eager.questioncloud.core.domain.creator.model.CreatorStatistics

interface CreatorStatisticsRepository {
    fun save(creatorStatistics: CreatorStatistics)

    fun saveAll(creatorStatistics: List<CreatorStatistics>)

    fun findByCreatorId(creatorId: Long): CreatorStatistics

    fun getForUpdate(creatorId: Long): CreatorStatistics

    fun findByCreatorIdIn(creatorIds: List<Long>): Map<Long, CreatorStatistics>

    fun addSalesCount(creatorId: Long, count: Int)

    fun deleteAllInBatch()
}
